### PR TITLE
Writing workshop

### PR DESCRIPTION
**Pull request by EdenComp**
# Description

This PR is the writing part of the workshop, including [README](./README.md) for workshop exercises and [SETUP](./SETUP.md) for installation instructions.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [x] I have added / updated tests (unit / functionals / end-to-end / ...)
- [x] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,124 @@
-# Workshop React TS
+# Workshop 1 - Kanban App in React
 
-This project is the React App used for the new React Typescript Workshop from PoC-Innovation!
+The goal of this workshop is to create a [Kanban-like](https://www.digite.com/kanban/what-is-kanban/) application using the React library in Typescript
```
Comment (RezaRahemtola) [inline README.md:3]: Not really a Kanban-like app anymore :joy:

File: README.md [hunk 2]
```diff
@@ -1,35 +1,124 @@
...
+<br/><br/>
+## Step 0: Initialization
 
-## How does it work?
+All the installation steps required to do the exercises are detailed in the [SETUP.md](./SETUP.md)
+<br/><br/>
+## Step 1: Create your first component
 
-This project aims to learn how to create a front-end with React using Typescript.
+`Components` are the core of React. They represent a single element in a web page, like a `text` box, a `button`, a `div` etc. All combined together, they create a fully working web page. These components are re-usable, they prevent you from code duplication.
```
Comment (RezaRahemtola) [inline README.md]: add some official documentation links, like https://reactjs.org/docs/components-and-props.html here for example

File: README.md [hunk 3]
```diff
@@ -1,35 +1,124 @@
...
 
-## Getting Started
+You will learn how to [create components](https://www.w3schools.com/react/react_components.asp) right now in the first step!
 
-### Installation
+> In this workshop, we recommend you to use `Functional Components`, instead of `Javascript classes`. [Here](https://djoech.medium.com/functional-vs-class-components-in-react-231e3fbd7108) is a little explanation of the differences.
 
-[Explain how to install all of the project's dependencies]
+In `src/App.ts`, create a component that will display the application title, which is just a string.
+In Typescript, we must specify the type of the return value of the function: for all TSX React components, you can use `React.Element`.
```
Comment (RezaRahemtola) [inline README.md]: JSX.Element

File: README.md [hunk 4]
```diff
@@ -1,35 +1,124 @@
...
 
-### Quickstart
+```tsx
+function Title(): React.Element {
```
Comment (RezaRahemtola) [inline README.md]: Same for the type

File: README.md [hunk 5]
```diff
@@ -1,35 +1,124 @@
...
+      <Title />
+    ...
+  );
+}
+```
+<br/><br/>
+## Step 2: Show content with Components from Material UI
 
-[Explain how to use this project]
+Now that you know how to create a component in React, let's get into Material UI.
```
Comment (RezaRahemtola) [inline README.md]: same, add a link to Material UI's website

File: README.md [hunk 6]
```diff
@@ -1,35 +1,124 @@
...
+Material UI is a library of pre-built components that you can use in your React app.
+Discover [here](https://dev.to/amberjones/5-delightful-things-about-material-ui-5402) the advantages of using a components library such as Material UI.
 
-## Our PoC team :heart:
+The goal of this task is to create the `Task` component: this component will represent a task with all its information, including:
```
Comment (RezaRahemtola) [inline README.md:44]: The goal of this **step** may be better to avoid repeating Task

File: README.md [hunk 7]
```diff
@@ -1,35 +1,124 @@
...
+- A due time
 
-Developers
-| [<img src="https://github.com/Samoten777.png?size=85" width=85><br><sub>Laure GAGNER</sub>](https://github.com/Samoten777) | [<img src="https://github.com/EdenComp.png?size=85" width=85><br><sub>Florian LAUCH</sub>](https://github.com/EdenComp) | [<img src="https://github.com/nicolasheude.png?size=85" width=85><br><sub>Nicolas HEUDE</sub>](https://github.com/nicolasheude)
-| :---: | :---: | :---: |
+To do so, you must use the [Material UI Components](https://mui.com/components/), such as [Box](https://mui.com/material-ui/react-box/) or [Typography](https://mui.com/material-ui/react-typography/).
 
+:bulb: We let you figure out which components you may use from the link below!
+
+> From now, we recommend you to create a `Components` folder in your `src` folder, and to create one file per component and naming it with the name of your component. For instance, the `Task` component should be located in `src/Components/App.tsx`.
```
Comment (RezaRahemtola) [inline README.md]: isn't this an old version ? Already did this comment too for the capitalized `Components` -> it should be `components`

File: README.md [hunk 8]
```diff
@@ -1,35 +1,124 @@
...
+<br/><br/>
+## Step 3: Show all the tasks from a list with props
+
+At this moment, you know how to create a single component and how to add it to your application. But what if you need to show a list of components that you cannot determine in advance? For example a list of tasks :thinking:
+
+To do so, you can create a `List` component that will show all the tasks iteratively. 
+
+> You can create a [Typescript type](https://www.typescriptlang.org/docs/handbook/basic-types.html) representing the tasks and their information, and a array to store them.
+
+You can then use [React Props](https://reactjs.org/docs/components-and-props.html) to pass variables from your main components to your children components, and iterate your `Tasks array` with the Javascript [map](https://reactjs.org/docs/lists-and-keys.html) function.
```
Comment (RezaRahemtola) [inline README.md]: if you talk specifically about props, maybe add a link precisely about it (another one or an anchor on this page like https://reactjs.org/docs/components-and-props.html#rendering-a-component)

File: README.md [hunk 9]
```diff
@@ -1,35 +1,124 @@
...
+
+:bulb: When you create a component through an iteration, don't forget to pass a unique `key` in component props!
+<br/><br/>
+## Step 4: Create interactive components using useState
+
+From now, you are able to create static components that doesn't interact with the user.
```
Comment (RezaRahemtola) [inline README.md]: don't instead of doesn't

File: README.md [hunk 10]
```diff
@@ -1,35 +1,124 @@
...
+We will now create dynamic components to allow the user to create a new task to the list, using [React hooks](https://reactjs.org/docs/hooks-intro.html).
```
Comment (RezaRahemtola) [inline README.md]: `to add a new task to the list` looks better.

`create [..] to the list` is strange :joy:

File: README.md [hunk 11]
```diff
@@ -1,35 +1,124 @@
...
+## Bonus
+
+Congratulations, you now have a functionnal Kanban style app to keep yourself organized in all your future projects!
+
+You can still enhance it's behaviour, here are some examples:
+
+- You can add a background to your application
+- You can add a `Delete` button on each task to delete it
+- You can add a Due time using Calendars
+- You can link your application to a `backend` and a `database` to properly store your tasks (EPyTodo ?)
```
Comment (RezaRahemtola) [inline README.md:111]: add some links in the bonus part, like some backend frameworks compatible with React (Next, Remix...)

**Review by RezaRahemtola — CHANGES_REQUESTED** [review]
Lot of small details to change, also try to format the steps like this https://github.com/PoCInnovation/Workshops/tree/master/software/18.Kafka
with Description/Tasks/Documentation (Validation if needed but useless for this workshop imo)

File: README.md (hunk 1 above)
Comment (EdenComp) [reply README.md:3]: Still a kanban - At least todo list aiming to create a kanban

File: README.md (hunk 2 above)
Comment (EdenComp) [reply README.md]: Done

**Review by RezaRahemtola — APPROVED** [review]
LGTM

**Pull request merged**